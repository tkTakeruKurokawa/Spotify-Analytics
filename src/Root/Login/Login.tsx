const authUri = "https://accounts.spotify.com/authorize";
const redirectUri = "http://localhost:3000/";
const scopes = [
  "user-read-recently-played",
  "user-top-read",
  "user-read-playback-position",
];

export const getUriHash = () => {
  return window.location.hash
    .substring(1)
    .split("&")
    .reduce((initial: any, item: string) => {
      const parts = item.split("=");
      initial[parts[0]] = decodeURIComponent(parts[1]);
      return initial;
    }, {});
};

const Login = () => {
  const loginUri = `${authUri}?client_id=${
    process.env.REACT_APP_SPOTIFY_CLIENT_ID
  }&redirect_uri=${redirectUri}&scope=${scopes.join(
    "%20"
  )}&response_type=token&show_dialog=true`;

  return (
    <div className="login">
      <h1>ログインする</h1>
      <a href={loginUri}>Spotifyにログイン</a>
    </div>
  );
};

export default Login;
