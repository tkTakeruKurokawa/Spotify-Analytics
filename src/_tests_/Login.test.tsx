import React from "react";
import renderer from "react-test-renderer";
import Login from "../Login";

it("Login test", () => {
  const component = renderer.create(<Login />);
  expect(component).toMatchSnapshot();
});
