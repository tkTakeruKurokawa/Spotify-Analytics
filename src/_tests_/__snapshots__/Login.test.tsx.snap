// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test 1`] = `
<div
  className="login"
>
  <h1>
    ログインする
  </h1>
  <a
    href="https://accounts.spotify.com/authorize?client_id=5c084d85cb73478fb7d26906f0214f78&redirect_uri=http://localhost:3000/&scope=user-read-recently-played%20user-top-read%20user-read-playback-position&response_type=token&show_dialog=true"
  >
    Spotifyにログイン
  </a>
</div>
`;
